plugins/modules/aci_ip_sla_monitoring_policy.py

```diff
@@ -0,0 +1,388 @@
+from ansible_collections.cisco.aci.plugins.module_utils.constants import HTTP_VERSIONS_MAPPING
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        tenant=dict(type="str", aliases=["tenant_name"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+        name=dict(type="str", aliases=["sla_policy"]),
+        description=dict(type="str"),
+        sla_type=dict(type="str", choices=["icmp", "tcp", "l2ping", "http"]),
+        sla_port=dict(type="int", aliases=["destination_port"]),
+        frequency=dict(type="int"),
+        multiplier=dict(type="int", aliases=["detect_multiplier"]),
+        request_data_size=dict(type="int"),
+        type_of_service=dict(type="int", aliases=["tos"]),
+        operation_timeout=dict(type="int"),
+        threshold=dict(type="int"),
+        traffic_class=dict(type="int"),
+        http_version=dict(type="str", choices=["1.0", "1.1"]),
+        http_uri=dict(type="str"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["tenant", "name"]],
+            ["state", "present", ["tenant", "name"]],
+        ],
+    )
+
+    tenant = module.params.get("tenant")
+    state = module.params.get("state")
+    description = module.params.get("description")
+    name = module.params.get("name")
+    sla_type = module.params.get("sla_type")
+    sla_port = module.params.get("sla_port")
+    frequency = module.params.get("frequency")
+    multiplier = module.params.get("multiplier")
+    request_data_size = module.params.get("request_data_size")
+    type_of_service = module.params.get("type_of_service")
+    operation_timeout = module.params.get("operation_timeout")
+    threshold = module.params.get("threshold")
+    traffic_class = module.params.get("traffic_class")
+    http_version = HTTP_VERSIONS_MAPPING.get(module.params.get("http_version"))
+    http_uri = module.params.get("http_uri")
+
+    aci = ACIModule(module)
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fvTenant",
+            aci_rn="tn-{0}".format(tenant),
+            module_object=tenant,
+            target_filter={"name": tenant},
+        ),
+        subclass_1=dict(aci_class="fvIPSLAMonitoringPol", aci_rn="ipslaMonitoringPol-{0}".format(name), module_object=name, target_filter={"name": name}),
+    )
+    aci.get_existing()
+
+    if state == "present":
+        if sla_port is not None and sla_type != "tcp":
+            aci.fail_json("Setting 'sla_port' is not allowed when 'sla_type' is not set to 'tcp'.")
+        if sla_type != "http" and request_data_size is not None:
+            aci.fail_json("Setting 'request_data_size' is not allowed when 'sla_type' is not set to 'http.")
```

Comment:
is this correct because in ui request data size seems to be accepted for icmp, l2ping and http?